### PR TITLE
Button vars

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: mobile/cmp/button/ButtonGroup.scss [hunk 1]
```diff
@@ -17,13 +17,13 @@ body.xh-app .xh-button-group .xh-button {
     background-color: var(--xh-border-color);
```
[the remark this answers is not in the document]
Comment (TomTirapani) [reply mobile/cmp/button/ButtonGroup.scss]: I can own up to this one! This is definitely weird... I think we want to use the `--xh-button-bg` here instead, and then the active color below, to be consistent with how we do it on desktop

File: kit/onsen/theme.scss [hunk 2]
```diff
@@ -30,11 +30,11 @@ body.xh-app.xh-mobile {
 
   .toolbar-button,
   .back-button {
-    color: var(--xh-toolbar-btn-color);
+    color: var(--xh-toolbar-button-bg);
```
[the remark this answers is not in the document]
Comment (peterjcoffman) [reply kit/onsen/theme.scss]: I don't even know if this does anything. It certainly doesn't set the color of the back button.

[the remark this answers is not in the document]
Comment (TomTirapani) [reply kit/onsen/theme.scss]: You're right Peter - thanks for catching. This is an outdated style from the earliest days of the mobile toolkit. We can remove this.

That said, I think we need to add a style that sets the `color` of `xh-button` within `xh-appbar` to `var(--xh-navbar-title-color)`. They are currently hardcoded to white, when they should be the same color as the app title.

[the remark this answers is not in the document]
Comment (peterjcoffman) [reply kit/onsen/theme.scss]: In this branch, the toolbox button is no longer hard-coded to white. It is `xh-button-text-color`.

I have changed it to `var(--xh-navbar-title-color)` and removed the toolbar-button and back-button styles.

**Comment by amcclain** [comment]
@TomTirapani I don't know if you will have a chance - pls. continue to prioritize immediate client work here - but would be great if you're able to take a look at this and see if we can get to a good state to merge.  This could slot in nicely to a v30 release since we'll be upping the major version for the DataView changes.

If not feasible, no worries - we can get it on the next go-round.

**Comment by TomTirapani** [comment]
Had a play with this by setting button vars to garish colors, and mostly seems to work as expected. :)

I found one issue - the app header buttons in mobile light theme appear invisible:

Screenshot 2020-02-18 at 14 27 17

Note that in dark theme the app header buttons are visible:

Screenshot 2020-02-18 at 14 27 25

**Comment by peterjcoffman** [comment]
@TomTirapani 

I'm a little confused with how we're handling the Mobile AppBar. Whether or this branch or develop, it looks quite different between my machine and production.

It looks as if xh-navbar-bg and xh-navbar-title-color variables are being ignored.

Here's what the dev tools shows for the xh-appbar-title on my machine:

image

And in production:

image

Why are they different?

**Comment by peterjcoffman** [comment]
@TomTirapani 

I believe the problem with invisible buttons was fixed by https://github.com/xh/hoist-react/commit/7cdbef53111e103c7bca4d81749545583846c9e4.

**Comment by peterjcoffman** [comment]
@TomTirapani I believe that https://github.com/xh/hoist-react/commit/7cdbef53111e103c7bca4d81749545583846c9e4 fixed the problem with the AppBar buttons being invisible. Would you take another look?

**Comment by TomTirapani** [comment]
Yep, looks good!

**Comment by peterjcoffman** [comment]
@TomTirapani @amcclain Shall we merge?

File: CHANGELOG.md [hunk 3]
```diff
@@ -93,6 +93,11 @@
   ([#1641](https://github.com/xh/hoist-react/pull/1641))
 * `AppOption` configs now accept an `omit` property for conditionally excluding options.
 
+### 💥 Breaking Changes
+
+* Several CSS variables related to buttons have been renamed for consistency.
+  ([#1568](https://github.com/xh/hoist-react/pull/1568))
```
Comment (amcclain) [inline CHANGELOG.md]: CHANGELOG version gotcha strikes again! 😉

**Comment by amcclain** [comment]
Thanks guys - merged and closed out this long-running cleanup! 👍

**Pull request merged**